File: pages/js/game.js
function newGame() {
  var templ = [
    '<h2>Make your choice</h2>',
    '<div class="box">',
      '<div>',
        '<input type="button" class="choice" name="choice" value="rock" onclick="sendChoice(\'rock\',\'{{=it.GameId}}\');">',
        '<input type="button" class="choice" name="choice" value="paper" onclick="sendChoice(\'paper\',\'{{=it.GameId}}\');">',
        '<input type="button" class="choice" name="choice" value="scissors" onclick="sendChoice(\'scissors\',\'{{=it.GameId}}\');">',
        '<input type="button" class="choice" name="choice" value="lizard" onclick="sendChoice(\'lizard\',\'{{=it.GameId}}\');">',
        '<input type="button" class="choice" name="choice" value="spock" onclick="sendChoice(\'spock\',\'{{=it.GameId}}\');">',
      '</div>',
    '</div>',
    "<div id='results'>",
    "</div>",
  ].join("\n");
  var tmp = $.post("/rpsls/new/")
  .done(function(data) {
    var response = JSON.parse(data);
    console.log(response);
    var tmp = doT.template(templ)(response);
    console.log(tmp);
    $('#content').empty().append(tmp);
  });
}

function sendChoice(selection, gameId) {
  var templ = [
      "<div class='result'>",
        "<h2><span id='result'>{{=it.LastGame}}</span></h2>",
      "</div>",
      "<h2>Game Stats</h2>",
      "<table class='table'>",
            "<tr>",
              "<th>Game ID</th>",
              "<td id='comment'>{{=it.GameId}}</td>",
            "</tr>",
            "<tr>",
              "<th>Games Played</th>",
              "<td class='comment'><span id='games'>{{=it.Games}}</span></td>",
            "</tr>",
            "<tr>",
              "<th>Player Score</th>",
              "<td class='comment'><span id='winsPlayer'>{{=it.PlayerWins}}</span></td>",
              "<td class='stat'><div class='bar'>",
              "<div class='p' style='display: in-line; width: {{=it.PlayerPercent}}%;'><span id='scorePlayer'>{{=it.PlayerPercent}}%</span></div>",
              "<div class='c' style='display: in-line; width: {{=it.ComputerPercent}}%;'><span id='scoreComputer'>{{=it.ComputerPercent}}%</span></div>",
              "<div class='t' style='display: in-line; width: {{=it.TiesPercent}}%;'><span id='scoreTies'>{{=it.TiesPercent}}%</span></div>",
              "</div></td>",
            "</tr>",
            "<tr>",
              "<th>Computer Score</th>",
              "<td class='comment'><span id='winsComputer'>{{=it.ComputerWins}}</span></td>",
            "</tr>",
            "<tr>",
              "<th>Ties</th>",
              "<td class='comment'><span id='winsTies'>{{=it.Ties}}</span></td>",
            "</tr>",
          "</table>",
      ].join("\n");
	var jqxhr = $.post( "/rpsls/result/"+gameId, {"choice": selection})
  .done(function(data) {
    var response = JSON.parse(data);
    var tmp = doT.template(templ)(response);
    $('#results').empty().append(tmp);
    $('#div:hidden:first').fadeIn("slow");
  });
}

File: logic/handler/game.go
package game

import (
	"math"
	"math/rand"
)

var (
	choice = [5]string {"rock", "paper", "scissors", "lizard", "spock"}
	moves = [10]Move {
		{Player1: "rock", Player2: "scissors", Flavor: "crushes"},
		{Player1: "rock", Player2: "lizard", Flavor: "crushes"},
		{Player1: "paper", Player2: "rock", Flavor: "covers"},
		{Player1: "paper", Player2: "spock", Flavor: "disproves"},
		{Player1: "scissors", Player2: "paper", Flavor: "cuts"},
		{Player1: "scissors", Player2: "lizard", Flavor: "decapitates"},
		{Player1: "lizard", Player2: "spock", Flavor: "poisons"},
		{Player1: "lizard", Player2: "paper", Flavor: "eats"},
		{Player1: "spock", Player2: "rock", Flavor: "vaporizes"},
		{Player1: "spock", Player2: "scissors", Flavor: "smashes"},
	}
)

type Move struct {
	Player1 string
	Player2 string
	Flavor string
}

type Game struct {
	GameId string
	Player []string
	LastGame string
	Computer []string
	Flavor []string
	PlayerWins int
	PlayerPercent float64
	ComputerWins int
	ComputerPercent float64
	Ties int
	TiesPercent float64
	Games int
}

func (g *Game) RunRound(p string) {
	c := computerChoice()
	g.Player = append(g.Player, p)

	g.Computer = append(g.Computer, c)

	for i := 0; i < 10; i++ {
		if p == c {
			g.Ties++
			g.Flavor = append(g.Flavor, "Ties")
			g.LastGame = p + " Ties " + c
			break
		}
		switch {
			case moves[i].Player1 == p && moves[i].Player2 == c:
				g.PlayerWins++
				g.Flavor = append(g.Flavor, moves[i].Flavor)
				g.LastGame = moves[i].Player1 + " " + moves[i].Flavor + " " + moves[i].Player2 + ", Player Wins"
				break
			case moves[i].Player1 == c && moves[i].Player2 == p:
				g.ComputerWins++
				g.Flavor = append(g.Flavor, moves[i].Flavor)
				g.LastGame = moves[i].Player1 + " " + moves[i].Flavor + " " + moves[i].Player2 + ", Computer Wins"
				break
		}
	}
	g.Games++
	//g.PlayerPercent = calcPercent(g.PlayerWins, g.Games)
	//g.ComputerPercent = calcPercent(g.ComputerWins, g.Games)
	//g.TiesPercent = calcPercent(g.Ties, g.Games)
	g.calcPercent()
}

func (g *Game) calcPercent() {
	if(g.Games == 0) {
		return
	}

	g.PlayerPercent = math.Floor(float64(g.PlayerWins) / float64(g.Games) * 100)
	g.ComputerPercent = math.Floor(float64(g.ComputerWins) / float64(g.Games) * 100)
	g.TiesPercent = math.Floor(float64(g.Ties) / float64(g.Games) * 100)
}

func computerChoice() string {
	return choice[rand.Intn(100) % 5]
}